WalletWasabi.Fluent/ViewModels/Wallets/LoadingViewModel.cs

```diff
@@ -100,12 +103,31 @@ private async Task LoadWalletAsync(bool isBackendAvailable)
 
 		await SetInitValuesAsync(isBackendAvailable).ConfigureAwait(false);
 
-		while (isBackendAvailable && RemainingFiltersToDownload > 0 && !_wallet.KeyManager.SkipSynchronization)
+		while (isBackendAvailable && RemainingFiltersToDownload > 0 && !_nbwsvm.Wallet.KeyManager.SkipSynchronization)
 		{
 			await Task.Delay(1000).ConfigureAwait(false);
 		}
 
-		await UiServices.WalletManager.LoadWalletAsync(_wallet).ConfigureAwait(false);
+		if (_nbwsvm.Wallet.State != WalletState.Uninitialized)
+		{
+			throw new Exception("Wallet is already being logged in.");
+		}
+
+		try
+		{
+			await Task.Run(async () => await Services.WalletManager.StartWalletAsync(_nbwsvm.Wallet));
```

Comment:
What is a reason of not awaiting the underlying task directly? Is `StartWalletAsync` a non ending method which must run on a background thread?

Reply:
@YohDeadfall this must run on a background thread, to prevent freezing the UI.